src/app/public/src/modules/clipboard/clipboard.service.ts

```diff
@@ -0,0 +1,104 @@
+// taken from ngx-clipboard
+// tslint:disable
+
+import { Inject, Injectable, Renderer } from '@angular/core';
+import { DOCUMENT } from '@angular/platform-browser';
+import { WINDOW } from 'ngx-window-token';
+
+@Injectable()
+export class ClipboardService {
+    private tempTextArea: HTMLTextAreaElement | undefined;
+    constructor(
+        @Inject(DOCUMENT) private document: any,
+        @Inject(WINDOW) private window: any,
```

Comment:
I think we can use our `windowRef` here instead of installing a library that does the same thing?
https://github.com/blackbaud/stache2/blob/master/src/app/public/src/modules/affix/affix-top.directive.ts#L9